airbyte-integrations/connectors/source-google-ads/source_google_ads/google_ads.py

```diff
@@ -99,8 +102,14 @@ def convert_schema_into_query(
         query_template = f"SELECT {fields} FROM {from_category} "
 
         if cursor_field:
-            # Fix issue 5411: Make date_start and date_end inclusive.
-            query_template += f"WHERE {cursor_field} >= '{from_date}' AND {cursor_field} <= '{to_date}' ORDER BY {cursor_field} ASC"
+            if from_category == 'click_view':
+                # Queries including ClickView must have a filter limiting the results to one day
+                # and can be requested for dates back to 90 days before the time of the request.
+                to_date = pendulum.parse(from_date).add(days=1).to_date_string()
+                query_template += f"WHERE {cursor_field} > '{from_date}' AND {cursor_field} <= '{to_date}' ORDER BY {cursor_field} ASC"
+            else:
+                # Fix issue 5411: Make date_start and date_end inclusive.
+                query_template += f"WHERE {cursor_field} >= '{from_date}' AND {cursor_field} <= '{to_date}' ORDER BY {cursor_field} ASC"
```

Comment:
Is it not possible to use the same `query_template` working only with `from_date` and `to_date` inside the if block?

Reply:
not possible @marcosmarxm , if block collect data for a day only but in else block end date depend on conversion_window so we can't use the same query. i tried but it failed and then read the google doc and change this logic.

Reply:
If I remember correctly, the `chunk_date_range `function creates several slices with a 1 month date range, for example:

from_date | to_date |
------------| ---------|
2020-03-17 | 2020-04-16 |
2020-04-17 | 2020-05-16 |
2020-05-17 | 2020-06-16 |
2020-06-17 | 2020-07-16 |
... | ... |

And in your case, you are requesting an entry for only one day in each month. 
In this case, I think it would be more appropriate to expand the `chunk_date_range `function with the ability to create a range with a date range of 1 day, rather than 1 month.

Reply:
sure @yevhenii-ldv,   , let me do it for one day in chunk_date_range range function, any other possibilities in the code where we can address this issue.

Reply:
@yevhenii-ldv any thoughts on below updated function
def chunk_date_range(start_date: str, conversion_window: int, field: str, report_name: str, end_date: str = None) -> Iterable[
    Mapping[str, any]]:
  
    intervals = []
    end_date = pendulum.parse(end_date) if end_date else pendulum.now()
    start_date = pendulum.parse(start_date)

    # As in to return some state when state in abnormal
    if start_date > end_date:
        return [{field: start_date.to_date_string()}]

    # applying conversion window
    start_date = start_date.subtract(days=conversion_window)

    # Each stream_slice contains the beginning and ending timestamp for a 24 hour period
    while start_date < end_date:
        intervals.append({field: start_date.to_date_string()})
        if report_name == 'click_view':
            start_date = start_date.add(days=1)
        else:
            start_date = start_date.add(months=1)

    return intervals